### PR TITLE
Dockerize Community Tests

### PR DESCRIPTION
**Pull request by karenc-bq**
### Summary

tests: integrate community tests into the new testing framework and update some tests.

### Description

- update some community tests
- dockerize all non-integration tests in `testsuite`

Other than the unintentional spacing modification, changes in `ConnectionRegressionTest` comes directly from https://github.com/mysql/mysql-connector-j/blob/release/8.0/src/test/java/testsuite/regression/ConnectionRegressionTest.java

### Additional Reviewers

@sergiyv-bitquill 
@ColinKYuen 
@crystall-bitquill

### CODE REVIEW

**Comment by karenc-bq** [comment]
Running the community tests locally using docker-compose instead of the new testing framework will probably cause the tests to fail because of my changes to BaseTestCase.

We will have to update relevant documentation in a separate PR.

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -11,8 +11,8 @@ on:
       - '**/*.md'
 
 jobs:
-  ubuntu-latest-aurora-build-and-run-tests:
-    name: 'Build Driver And Run Tests'
+  ubuntu-latest-aurora-build:
```
Comment (ColinKYuen) [inline .github/workflows/main.yml:14]: Building then not doing anything with it. Might as well remove this job

Comment (karenc-bq) [reply .github/workflows/main.yml:14]: I thought we want to reuse the artifacts built during test during release so we don't build it multiple times?

Comment (ColinKYuen) [reply .github/workflows/main.yml:14]: Each job runs with a different runner. We're setting Java up and recloning the repo in each repo. Also, the test dockers compile again. I tried to get caching to work within the containers but wasn't able to get it to work.

Comment (karenc-bq) [reply .github/workflows/main.yml:14]: Okay I will just remove it then

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -144,3 +116,30 @@ jobs:
...
+        with:
+          fetch-depth: 50
+      - name: 'Set up JDK 8'
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
+      - name: 'Run Community Tests'
+        run: |
+          ./gradlew --no-parallel --no-daemon test-community-docker
+        env:
```
Comment (ColinKYuen) [inline .github/workflows/main.yml:134]: Do we need these env variables for the community tests?

**Review by ColinKYuen — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

File: gradle.properties [hunk 3]
```diff
@@ -49,8 +48,8 @@ com.mysql.cj.build.licenseType=GPLv2
 
 systemProp.user.timezone=Africa/Lagos
 
-systemProp.com.mysql.cj.testsuite.url=jdbc:mysql://localhost:3306/test?serverTimezone=Africa/Lagos&user=root&password=root
-systemProp.com.mysql.cj.testsuite.url.openssl=jdbc:mysql://localhost:3307/test?serverTimezone=Africa/Lagos&user=root&password=root
+systemProp.com.mysql.cj.testsuite.url=jdbc:mysql://domain:port/test?serverTimezone=Africa/Lagos&user=root&password=root&allowPublicKeyRetrieval=true
```
Comment (sergiyv-bitquill) [inline gradle.properties:51]: {domain}:{port} 
to make it visible that these are parameters?

File: gradle.properties [hunk 4]
```diff
@@ -49,8 +48,8 @@ com.mysql.cj.build.licenseType=GPLv2
...
+systemProp.com.mysql.cj.testsuite.url.openssl=jdbc:mysql://domain:port/test?serverTimezone=Africa/Lagos&user=root&password=root&allowPublicKeyRetrieval=true
```
Comment (sergiyv-bitquill) [inline gradle.properties:52]: likewise

**Review by sergiyv-bitquill — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

**Review by sergiyv-bitquill — APPROVED** [review]

**Review by ColinKYuen — APPROVED** [review]

**Pull request merged**